File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        common_index = 0; strs.sort(key=len)
        for i in range(len(strs[0])):
            check = True
            char = strs[0][i]
            for j in strs:
                if j[i] != char:
                    check = False
                    break
            if check:
                common_index += 1
            else:
                break
        
        return strs[0][:common_index]

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs, (a, b) -> a.length() - b.length());

        int prefix = strs[0].length();

        for (String s: strs) {
            for (int i = 0; i < prefix; i++) {
                if (s.charAt(i) != strs[0].charAt(i)) {
                    prefix = i;
                    break;
                }
            }
        }

        return strs[0].substring(0, prefix);
    }
}